src/Microsoft.Extensions.DiagnosticAdapter/Internal/ProxyTypeEmitter.cs

```diff
@@ -101,10 +101,17 @@ private static bool VerifyProxySupport(ProxyBuilderContext context, Tuple<Type,
             }
 
             ProxyTypeCacheResult cacheResult;
+            var verificationResult = new VerificationResult();
             if (context.Cache.TryGetValue(key, out cacheResult))
             {
+                bool isError = cacheResult.IsError;
+                if (!isError && !context.Visited.ContainsKey(key))
+                {
+                    context.Visited.Add(key, verificationResult);
+                }
```

Comment:
I don't think this condition is correct. We should always add the verification result here. 

We already know that `context.Visited.ContainsKey(key)` returns false (Line 97). 

We also want to maintain the postcondition that everything we look at is in `context.Visited`, so it should go in there regardless of whether it's an error.